src/main/java/de/adesso/kicker/season/service/SeasonService.java

```diff
@@ -0,0 +1,135 @@
+package de.adesso.kicker.season.service;
+
+import de.adesso.kicker.match.persistence.Match;
+import de.adesso.kicker.match.service.MatchService;
+import de.adesso.kicker.season.persistence.*;
+import de.adesso.kicker.user.persistence.User;
+import de.adesso.kicker.user.service.UserService;
+import de.adesso.kicker.user.trackedstatistic.persistence.TrackedStatistic;
+import de.adesso.kicker.user.trackedstatistic.service.TrackedStatisticService;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class SeasonService {
+
+    private final SeasonRepository seasonRepository;
+
+    private final UserService userService;
+
+    private final TrackedStatisticService trackedStatisticService;
+
+    private final MatchService matchService;
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    private final Logger logger = LoggerFactory.getLogger(SeasonService.class);
+
+    private static final int SEASON_OFFSET = 1;
+
+    @Transactional
+    @Scheduled(cron = "0 0 1 1 1,4,7,10 ?")
+    void finishSeason() {
+        var seasonName = getSeasonName();
+        backUpOldSeason(seasonName);
+        deleteOldSeason();
+        seasonEndedEvent(seasonName);
+    }
+
+    private String getSeasonName() {
+        var seasonNumber = seasonRepository.count() + SEASON_OFFSET;
+        return String.format("Season %d", seasonNumber);
+    }
+
+    private void backUpOldSeason(String seasonName) {
+        var seasonMatches = copyAllMatches();
+        var seasonStatistics = copyAllStatistics();
+
+        var season = new Season(seasonName, seasonMatches, seasonStatistics);
+        seasonRepository.save(season);
+        logger.info("{} has been saved", seasonName);
+    }
+
+    private void deleteOldSeason() {
+        matchService.deleteAll();
+        logger.info("All matches have been deleted");
+
+        userService.deleteAllStatistics();
+        logger.info("All statistics have been deleted");
+    }
+
+    private void seasonEndedEvent(String seasonName) {
```

Comment:
I'd like to have this named `sendSeasonEndedEvent`

Reply:
Also may i'm overlooking it but is there any listener for this event yet?

Reply:
I didn't implement any listener's because that wan't of my concern.